src/shared/serialize.h

```diff
@@ -27,6 +27,9 @@ static inline int serialize_bool_elide(FILE *f, const char *key, bool b) {
         return b ? serialize_item(f, key, yes_no(b)) : 0;
 }
 
+#define serialize_item_format_tristate(file, field, value) \
+        serialize_item_format(file, field, "%i", value)
```

Comment:
should probably suppress the serialization if < 0, i.e. in the "unset", "invalid", "undefined" case